Refactor degree qualifications to support multiple suffixes and align better to Bologna

jelmerderonde opened the pull request:
This PR attempts to address #298 by restructuring how we represent academic qualifications to better align with international standards and the Bologna Process framework.

### Changes
- **Renamed** `qualificationAwarded` → `degreeAwarded` for clearer semantics
- **Simplified** degree types to core Bologna cycles:
  - `associate_degree` (EQF level 5)
  - `bachelor` (EQF level 6) 
  - `master` (EQF level 7)
  - `doctoral` (EQF level 8)
  - `none`
- **Added** new `degreeDesignations` array field to capture (if needed more than one) field-specific designations like "of Arts", "of Sciences", "of Engineering"

### Rationale
The Bologna framework focuses on degree cycles rather than prescribing specific designations. By separating the degree level (`degreeAwarded`) from field designations (`degreeDesignations`), we support international variations in naming conventions and enable interdisciplinary programs with multiple designations

### Breaking changes
- Field rename: `qualificationAwarded` → `degreeAwarded`
- Enum values changed (e.g., `ba`/`bsc` → `bachelor` + designation)
- Query parameter names updated in all collection endpoints

hamrt reviewed (commented):
Great work on simplifying the degree structure. Do we also want to standardize on the type of designation or leave that fully open for the time beeing? 

By switching to degree we are leaving out the VET sector. I think we need a deeper discussion on that, since qualifications are very much part of the way in which programmes are created in the VET sector.

VET has at least the following types:
1. Diploma / Vocational Diploma
Common term for full VET qualifications at upper secondary (ISCED 3–4) or post-secondary non-tertiary (ISCED 5).
E.g. Diploma in Automotive Technology.

2. Certificate
3. Smaller VET qualifications, often modular or stepwise (aligned with national qualification frameworks).


So we might want to rethink the naming convention: 
	•	In HE systems: use Degree Awarded.
	•	In VET systems: use Qualification Awarded with the specific title (Diploma, Certificate, Professional Bachelor).
	•	If you want a generic umbrella across both: we should keep “Qualification” (this is exactly why ELM and Europass prefer “qualification” as the base concept).

Visje01 commented:
I have looked at the proposed change and I think it is better not to amend this. If you look at how qualification is described in the diploma supplement, it is an umbrella term for all kinds of qualifications, of which degree is one. In other words, the format is broader than just the Dutch context. And broader than the higher education context. A few excerpts from the official instructions:
_The purpose of the Diploma Supplement is to provide sufficient independent data to improve the international
‘transparency’ and fair academic and professional recognition of qualifications (diplomas, degrees, certificates
etc.)._
The complete diploma supplement also refers to qualification in all fields. At the same time, this also covers the option of diploma as a type of qualification as used within VET.
At the same time, it is a good idea to make an adjustment, but not in the field names, but in the enumeration of the existing field qualificationAwarded. The current enumeration does indeed meet the intended use (VET is right about that). It would be better to create an enumeration list with the options bachelor of arts, bachelor of science, diploma, certificate, etc. There is a central register of internationally recognised degrees. This would improve the content and resolve the issue raised by VET.
The diploma supplement also makes a clear distinction between the qualification and the level by placing them in separate fields. The current ooapi version does the same.
In my opinion, it is not necessary to add a separate field unless the enumeration list becomes too long and you want to split the options into two fields, but this would make it more complex to submit data.

jelmerderonde commented:
@hamrt and @Visje01, I've tried updating the PR to satisfy your feedback:

- Renamed the fields back to use `qualification` instead of `degree`.
- Added some enums for VET and general use: `diploma`, `vocational_diploma` and `certificate`.

hamrt reviewed (commented):
Should de documentation also mention that qualificationAwarded and qualificationDesignations are working on tandem or is that self-explanitory?

rrutte commented on v6/schemas/ProgrammeProperties.yaml:
```diff
@@ -54,6 +54,12 @@ properties:
     $ref: '../enumerations/qualificationAwarded.yaml'
+  qualificationDesignations:
+      type: array
+      description: Academic field designations that specify the discipline area of the degree (e.g., "of Arts", "of Sciences", "of Engineering"). Multiple designations may apply to interdisciplinary programs.
```
Wouldn't it be better to have an array of language typed strings. Especially in vocational education it can be very helpful to have both the dutch/german/french/italian designation next to an english one.

hamrt replied:
I'm not sure how we would go about achieving that in tandem with having a enum list. Maybe we should shelve that discussion and add it as a separate new issue, so this version can be committed?

rrutte reviewed (commented):
See review comment on qualificationDesignations
